keps/sig-api-machinery/2839-in-use-protection/README.md

```diff
@@ -303,10 +311,38 @@ Therefore, liens field isn't defined as a subresource and providing a separate R
 
 #### Unresolved issues
 
-- [ ] Choice of an advisory feature
-- [ ] Subresource (Is the conclusion that it is needed but will be added later OK?)
-- [ ] Compatibility to existing deletion flows for existing tools (Is current behavior OK and current description enough?)
-- [ ] Name of the field (Is `liens` OK?)
+- Decision: When you delete a namespace and an object in that namespace has a lien:
+  - Does the namespace and object get deleted anyway?
+    - [x] Yes.
+    - [ ] No. See below cases:
+      - Does the namespace deletion get blocked (i.e., the delete request fails) if any object in the namespace has a lien?
+        - No.
+      - Does the other objects in the namespace get deleted, but that not object, preventing the namespace deletion from completing?
+        - [x] No. Namespace lifecycle cleanup will not honor liens. if you delete the namespace, the content will be removed.
+	    - [ ] Yes. See below case:
+          - Does the namespace deletion not proceed until all objects in the namespace are free of liens?
+            - [x] No. Namespace lifecycle cleanup will not honor liens. If you delete the namespace, the content will be removed.
+	        - [ ] ~~Yes. Namespace lifecycle cleanup will honor liens. If you delete the namespace, it won't be deleted until all the content will be deleted.~~
+	        - [ ] Partially yes. Namespace lifecycle cleanup will honor liens. If you delete the namespace, it won't start deleting the content with liens until all the content without liens will be deleted (Current KEP's description (Also see [Namespace Deletion](#namespace-deletion))).
+- Name:
+    - [x] "liens" is short and precise, but an unusual word
+    - [ ] Other candidates: InUse, deleteInhibitors, hold, lease, claim, deletionBlockers, protections, guards
+- Subresource:
+  - Do you need a special permission to add a lien?
+    - [x] Yes. Liens can only be changed via a call to resource/liens subresource. Create/update permission for the resource/liens subresource is required.
```

Comment:
I may be asking a stupid question, though.
Can we just add liens field as a subresource in metadata, technically?

(I was thinking that liens also need to be subresourced in the same way to "subresource finalizer" KEP. And if so, it would be difficult to rely on the delete permission to liens subresource).